README.md

```diff
@@ -1,117 +1,209 @@
 
 > Looking for previous versions of this code sample? Check out the tags on the [releases](../../releases) GitHub page.
 
 For more information about how the protocols work in this scenario and other scenarios, see the [Authentication Scenarios for Azure AD](https://azure.microsoft.com/documentation/articles/active-directory-authentication-scenarios/) document.
 
-## How To Run This Sample
+![Overview](./ReadmeFiles/topology.png)
+
+### Scenario
 
->[!Note] If you want to run this sample on **Azure Government**, navigate to the "Azure Government Deviations" section at the bottom of this page.
+This sample demonstrates a multi-tenant .NET Web App (MVC) application signing in users from multiple Azure AD tenants and calling [Microsoft Graph](https://graph.microsoft.com).
+
+1. The web app allows the tenant admins to sign-up (and provision this app in their tenant) by signing them in using the [ASP.Net OpenID Connect OWIN middleware](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet).
+1. The users of these signed-up tenants can then sign-in themselves and create a Todo list for themselves.
+1. The application also uses the [Active Directory Authentication Library (ADAL) for .NET](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet) library to obtain a JWT access token from Azure Active Directory (Azure AD) for [Microsoft Graph](https://graph.microsoft.com).
+1. The access token is used as a bearer token to authenticate the user when calling [Microsoft Graph](https://graph.microsoft.com) to fetch the signed-in user's details.
+
+## How to run this sample
+
+To run this sample, you'll need:
+
+- [Visual Studio 2017](https://aka.ms/vsdownload)
+- An Internet connection.
+- An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, see [How to get an Azure AD tenant](https://azure.microsoft.com/en-us/documentation/articles/active-directory-howto-tenant/)
+- A user account in your Azure AD tenant. This sample will not work with a Microsoft account (formerly Windows Live account). Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a Microsoft account and have never created a user account in your directory before, you need to do that now.
+
+>[!Note] If you want to run this sample in a **National Cloud**, navigate to the ["National Cloud Deviations"](#national-cloud-deviations) section at the bottom of this page.
```

Comment:
Microsoft has three national cloud environments.

1. Microsoft Cloud for US Government
2. Microsoft Cloud Germany
3. Azure and Office 365 operated by 21Vianet in China

If you want to run this sample in one of the ** Microsoft National Cloud** i.e. , navigate to the [" National Cloud Deviations"](#national-cloud-deviations) section at the bottom of this page.